### PR TITLE
Adds test scenarios for Basic01

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

This PR adds test scenarios for Basic01. The scenarios are based on the expected outcome of https://github.com/zonemaster/zonemaster/pull/1257.

### CODE REVIEW

File: test-zone-data/Basic-TP/basic01/COMMON.basic01.xa.zone [hunk 1]
```diff
@@ -0,0 +1,34 @@
+; Shared zone files for granparent zones when "parent"
+; is to be delegated in the standard way or overwridden
+; in basic01.cfg file.
```
Comment (marc-vanderwal) [inline test-zone-data/Basic-TP/basic01/COMMON.basic01.xa.zone:3]:
```suggestion
; Shared zone files for grandparent zones when "parent"
; is to be delegated in the standard way or overridden
; in basic01.cfg file.
```

File: test-zone-data/Basic-TP/basic01/basic01.xa.zone [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+		NS	ns1
+		NS	ns2
+
+ns1		A	127.12.1.21
+ns1		AAAA	fda1:b2:c3:0:127:12:1:21
+ns2		A	127.12.1.22
+ns2		AAAA	fda1:b2:c3:0:127:12:1:22
+
+; Name server names that holds no zones. Just provids addresses.
```
Comment (marc-vanderwal) [inline test-zone-data/Basic-TP/basic01/basic01.xa.zone:19]:
```suggestion
; Name server names that holds no zones. Just provides addresses.
```

**Review by marc-vanderwal — COMMENTED** [review]
Looks fine to me. I just see a few typos.

**Review by marc-vanderwal — DISMISSED** [review]

**Comment by matsduf** [comment]
@tgreenx, can you approve this?

File: docs/public/specifications/test-zones/Basic-TP/basic01.md [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+* B01_NO_CHILD
+* B01_PARENT_DISREGARDED
+* B01_PARENT_FOUND
+* B01_PARENT_NOT_FOUND
+* B01_PARENT_UNDETERMINED
+* B01_ROOT_HAS_NO_PARENT
+* B01_SERVER_ZONE_ERROR
+
+
+B01_CHILD_FOUND, B01_PARENT_DISREGARDED
```
[the remark this answers is not in the document]
Comment (matsduf) [reply docs/public/specifications/test-zones/Basic-TP/basic01.md]: Yes, forgot to remove it. Removed now.

File: test-zone-data/Basic-TP/basic01/profile.json
[the remark this answers is not in the document]
Comment (matsduf) [reply test-zone-data/Basic-TP/basic01/profile.json:1]: I see this as a temporary solution. I want to have one shared profile under "COMMON", and in another PR I do add that, but while waiting for that to be merged I created this profile file to make it work.

I have reduced it, but kept all Basic related messages plus SYSTEM.

[the remark this answers is not in the document]
Comment (matsduf) [reply test-zone-data/Basic-TP/basic01/profile.json:1]: Yes, the purpose is to raise any relevant message from DEBUG to INFO to make it possible to have all relevant messages printed out without have a lot of other messages. That is the only purpose of the profile here.

**Pull request merged**